_posts/en/2024-10-09-wallet-integration-guide.md

```diff
@@ -0,0 +1,600 @@
+  "complete": true
+}
+
+bitcoin-cli -regtest -rpcwallet=test submitpackage '["020000000001013b64742e658ff39da0288a5010f094a9e94e6ce3a3ce2ed84b74babc017aea490000000000fdffffff0111ff94000000000016001400991cdadccdf30cb5a04663b0371cb433a095b4024730440220488d98ad79495276bb4cdda4d7c62292043e185fa705d505c7dceef76c4b61d30220567243245416a9dd3b76f3d94bfd749e0915929226ba079ec918f6675cbfa3950121030af1fadce80bcb8ba614634bc82c71eea2ed87a5692d3127766cc896cef1bdb100000000", "0200000000010159eb4ce4b998b40fdf81395f37b2317a632c38c06f257747b09c027ad84671f10000000000fdffffff01405489000000000016001400991cdadccdf30cb5a04663b0371cb433a095b40247304402205d086fa617bdbf5a3df3a15cc9a927ad884c714d46d9ef6762ad2fa6a259740c022032c60b4fe5d533d990489c27dc3283d8b3999b97f6c12986ac8159b92cb6de820121020797cc343a24dfe49c7ee9b94bf3daaf15308d8c12e3f0f7e102b95ee55f939f00000000"]'
+{
+  "package_msg": "success",
+  "tx-results": {
+    "fe15d23f59537d12cddf510616397b639a7b91ba2f846c64533e847e53d7c313": {
+      "txid": "f17146d87a029cb04777256fc0382c637a31b2375f3981df0fb498b9e44ceb59",
+      "vsize": 110,
+      "fees": {
+        "base": 0.00001000,
+        "effective-feerate": 0.03480113,
+        "effective-includes": [
+          "fe15d23f59537d12cddf510616397b639a7b91ba2f846c64533e847e53d7c313",
+          "256cebd037963d77b2692cdc33ee36ee0b0944e6b9486a6aaad0792daa0f677c"
+        ]
+      }
+    },
+    "256cebd037963d77b2692cdc33ee36ee0b0944e6b9486a6aaad0792daa0f677c": {
+      "txid": "858fe07b01bc7c1c1dda50ba16a33b164c0bc03d0eff8f9546558c088e087f60",
+      "vsize": 110,
+      "fees": {
+        "base": 0.00764625,
+        "effective-feerate": 0.03480113,
+        "effective-includes": [
+          "fe15d23f59537d12cddf510616397b639a7b91ba2f846c64533e847e53d7c313",
+          "256cebd037963d77b2692cdc33ee36ee0b0944e6b9486a6aaad0792daa0f677c"
+        ]
+      }
+    }
+  },
+  "replaced-transactions": [
+    "bf9164db69d216da4af6d9c720a0cec6d7e0bafb1702fdf8c2cd5606101576de",
+    "6c2f4dec614c138703f33e6a5c215112bad4cf79593e9757105e09b09bf3e2de"
+  ]
+}
+
+```
+
+## Pay To Anchor (P2A)
+
+[Anchors][topic anchor outputs] are defined as an output that is added solely to allow a child
+transaction to CPFP that transaction. Ideally these outputs are just at “dust”
+satoshi values, and are immediately spent.
+
+A new output script type, [Pay To Anchor (P2A)][topic ephemeral anchors], has been added, which allows for an optimized “keyless”
+version of anchors. The output script is “OP_1 <4e73>”, which requires no witness
+data to spend, meaning a fee reduction compared to existing anchor outputs and
+allows anyone to spend it.
+
+P2A can be used independently from TRUC transactions or 1P1C packages. A transaction with a P2A output but no child can be broadcast, though the output is trivially spendable.
```

Comment:
"Similarly, packages and TRUC transactions do not need to have P2A outputs in order to use the new fee-bumping features."